pkg/templates/pdb/template.go

```diff
@@ -0,0 +1,101 @@
+package pdb
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/extract"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/pdb/internal/params"
+	pdbV1 "k8s.io/api/policy/v1"
+)
+
+const (
+	templateKey = "pod-disruption-budget"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Pod Disruption Budget not specified",
+		Key:         templateKey,
+		Description: "Flag DeploymentLike objects having no \"Pod Disruption Budget\" set in case if the replica count is more than the minimum specified value",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.DeploymentLike,
+				objectkinds.PodDisruptionBudget},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				var results []diagnostic.Diagnostic
+
+				replicas, _ := extract.Replicas(object.K8sObject)
+				if int(replicas) > p.MinReplicas {
+					namespaceForDeploymentLike := object.K8sObject.GetNamespace()
+					PDBFound := false
+					maxUnavailable := false
+					labels := extract.Labels(object.K8sObject)
+
+					pdbMaxUnavailable, pdbSelectors := gatherPDBs(lintCtx, namespaceForDeploymentLike)
+					if !(pdbSelectors == nil) {
```

Comment:
Usual style here is check for the condition and return nil if allowed to be empty, error otherwise.